thirdparty/patches/glog-suffix-log.patch

```diff
@@ -0,0 +1,28 @@
+diff --git src/logging.cc src/logging.cc
+--- src/logging.cc
++++ src/logging.cc
+@@ -1135,7 +1135,8 @@ void LogFileObject::Write(bool force_flush,
+                     << setw(2) << tm_time.tm_min
+                     << setw(2) << tm_time.tm_sec
+                     << '.'
+-                    << GetMainThreadPid();
++                    << GetMainThreadPid()
++                    << ".log";
+     const string& time_pid_string = time_pid_stream.str();
+ 
+     if (base_filename_selected_) {
+@@ -1146,6 +1147,14 @@ void LogFileObject::Write(bool force_flush,
+         return;
+       }
+     } else {
++      // NOTE(lingxuan.zlx): Why we need do this? Raylet process will never
++                       // call StartRayLog, so all log items created by RayLog was
++      // erroneously rewritten to a special file in the tmp directory.
++      // To avoid this, we need to output the log directly to standard
++                       // output so that the log is redirected to either raylet.out or
++                       // raylet.err.
++      std::cout << std::string(message, message_len) << std::flush;
```

Comment:
I am a little confused about this change (maybe I am lack of context). But, why don't we just make raylet call `StartRayLog` instead of this patch? Also, isn't it affecting other components that actually call `StartRayLog`?

Reply:
> I am a little confused about this change (maybe I am lack of context). But, why don't we just make raylet call `StartRayLog` instead of this patch? Also, isn't it affecting other components that actually call `StartRayLog`?

Actually raylet did call StartRayLog already, but it's none of severity has been mapped to pass a log dir with empty string.  If we give a non-empty log dir in raylet so that stderr and stdout of raylet may not be redirected to raylet.out/raylet.err. (it becomes normal log filename pattern like raylet.${TIMESTAMP}.${PID}.log)

Reply:
Imo, having a little longer raylet name is better than adding this patch? This is really ugly. Btw, it seems like you removed this patch already (and the log file name is still raylet.out). Did you figure out something?

Reply:
When log_dir is non-empty string, we will attach all logs to glog default logger, so they can be sinked to log files. 

And for empty log_dir, actually glog can not write anything to logfile.

Above two cases:
1. non-empty log dir -> glog defaut logger -> log file ( filename format ${appname}.${timestamp}.${pid}.log
2. empty log dir -> stdout logger (we defined it in src/ray/util/logging.cc), but python/ray/node.py::start_ray_processes will redirect std out and std err of raylet/gcs to specific directory and name, which is why we see their log informations in raylet.out/raylet.err.